sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -189,6 +189,7 @@ object AnsiTypeCoercion extends TypeCoercionBase {
         }
 
       case (DateType, TimestampType) => Some(TimestampType)
```

Comment:
shall we support Date -> Timestamp NTZ? in case we have such functions in the future.

Reply:
Well we can have it but it is not related to this PR and there is no test case for the change